Two sample t-test for lung cancer incidence in the US by gender

In [4]:
# Dependencies
import requests
import numpy as np
import scipy.stats as stats
from pprint import pprint

In [11]:
# url to our data
url = "http://127.0.0.1:5000/raw-lung-cancer-data"
# Make request and store response
response = requests.get(url)
# JSON-ify response
response_json = response.json()
# Print first article
pprint(response_json)

[{'DataValue': '57',
  'DataValueAlt': '57',
  'DataValueType': 'Average Annual Number',
  'LocationAbbr': 'DC',
  'LocationID': '11',
  'Question': 'Cancer of the lung and bronchus, incidence',
  'Stratification1': 'White, non-Hispanic',
  'StratificationCategory1': 'Race/Ethnicity',
  'StratificationCategoryID1': 'RACE',
  'StratificationID1': 'WHT',
  'YearEnd': '2019',
  'YearStart': '2015',
  'id': 1},
 {'DataValue': '2023',
  'DataValueAlt': '2023',
  'DataValueType': 'Average Annual Number',
  'LocationAbbr': 'CO',
  'LocationID': '8',
  'Question': 'Cancer of the lung and bronchus, incidence',
  'Stratification1': 'White, non-Hispanic',
  'StratificationCategory1': 'Race/Ethnicity',
  'StratificationCategoryID1': 'RACE',
  'StratificationID1': 'WHT',
  'YearEnd': '2019',
  'YearStart': '2015',
  'id': 2},
 {'DataValue': '8481',
  'DataValueAlt': '8481',
  'DataValueType': 'Average Annual Number',
  'LocationAbbr': 'CA',
  'LocationID': '6',
  'Question': 'Cancer of the lung and

In [16]:
# obtain the data for the population and two samples

female_samples = []
male_samples = []

# loop though json to get specific data
for x in response_json:
    if x['LocationAbbr'] == "US" and x['Stratification1'] == "Overall":
        us_overall = float(x['DataValue'])
    elif x['LocationAbbr'] != "US" and x['Stratification1'] == "Female":
        female_samples.append(float(x['DataValue']))
    elif x['LocationAbbr'] != "US" and x['Stratification1'] == "Male":
        male_samples.append(float(x['DataValue']))

print(female_samples)
print(male_samples)
print(us_overall)


[8481.0, 412.0, 1302.0, 1247.0, 1986.0, 1720.0, 158.0, 173.0, 1409.0, 715.0, 6270.0, 4713.0, 3119.0, 2871.0, 2027.0, 468.0, 328.0, 539.0, 2816.0, 481.0, 1583.0, 4865.0, 1190.0, 3988.0, 243.0, 975.0, 2148.0, 1162.0, 8468.0, 2650.0, 489.0, 2672.0, 379.0, 3910.0, 2290.0, 257.0, 3162.0, 5372.0, 612.0, 2664.0, 305.0, 2345.0, 1470.0, 7334.0, 365.0, 1858.0, 1409.0, 606.0, 938.0, 150.0, 1893.0]
[1292.0, 201.0, 141.0, 1160.0, 413.0, 2259.0, 1626.0, 2023.0, 8424.0, 269.0, 5290.0, 2573.0, 650.0, 1664.0, 2208.0, 2816.0, 2432.0, 1364.0, 562.0, 2269.0, 1965.0, 465.0, 6901.0, 2951.0, 719.0, 4104.0, 5541.0, 256.0, 151.0, 310.0, 2248.0, 1349.0, 350.0, 3351.0, 373.0, 502.0, 2832.0, 4739.0, 495.0, 2853.0, 1516.0, 495.0, 1884.0, 9193.0, 7291.0, 1891.0, 4434.0, 989.0, 1112.0, 459.0, 3717.0]
224257.0


In [17]:
stats.ttest_ind(female_samples, male_samples, equal_var=False)

Ttest_indResult(statistic=-0.2810449172608385, pvalue=0.7792577059693152)

The p-value (pvalue=0.7792577059693152) is over 0.05. Therefore, the difference between the male and female data samples is not significant.

*If a p-value reported from a t test is less than 0.05, then that result is said to be statistically significant. If a p-value is greater than 0.05, then the result is insignificant.